File: src/main/java/no/nav/veilarboppfolging/client/veilarbarena/VeilarbArenaOppfolgingsStatus.java
package no.nav.veilarboppfolging.client.veilarbarena;

import lombok.Data;
import lombok.experimental.Accessors;

import javax.annotation.Nullable;
import java.time.LocalDate;

/**
 * Har ikke feltene "hovedmaal" og "kvalifiseringsgruppe" (men servicegruppe er egentlig kvalifiseringsgruppe) som {@link no.nav.veilarboppfolging.client.veilarbarena.VeilarbArenaOppfolgingsBruker} har
 * @see no.nav.veilarboppfolging.client.veilarbarena.VeilarbArenaOppfolgingsBruker
 */
@Data
@Accessors(chain = true)
public class VeilarbArenaOppfolgingsStatus {
    String rettighetsgruppe;
    String formidlingsgruppe;
    String servicegruppe;
    String oppfolgingsenhet;
    LocalDate inaktiveringsdato;
    @Nullable
    Boolean kanEnkeltReaktiveres;
}

File: src/main/kotlin/no/nav/veilarboppfolging/client/veilarbarena/VeilarbArenaOppfolgingsStatus.kt
package no.nav.veilarboppfolging.client.veilarbarena

import java.time.LocalDate


/**
 * Har ikke feltene "hovedmaal" og "kvalifiseringsgruppe" (men servicegruppe er egentlig kvalifiseringsgruppe) som [VeilarbArenaOppfolgingsBruker] har
 * @see VeilarbArenaOppfolgingsBruker
 */
data class VeilarbArenaOppfolgingsStatus(
    val rettighetsgruppe: String?,
    val formidlingsgruppe: String?,
    val servicegruppe: String?,
    val oppfolgingsenhet: String?,
    val inaktiveringsdato: LocalDate? = null,
    val kanEnkeltReaktiveres: Boolean? = null,
)
